code/number-theory/Sieve.h

```diff
@@ -0,0 +1,23 @@
+const int N = 1e6;
+int lpd[N];
+vector<int> primes ; 
+vector<int> factorize(int n) {
+   vector<int> ans;
+   while (n != 1) {
+      ans.push_back(lpd[n]);
+      n/= lpd[n];
```

Comment:
In your place, I would write this as:
```cpp
while (n != 1) ans.push_back(lpd[n]), n/= lpd[n];
```
This takes a lot less space on paper (we tend to have more horizontal space than vertical space).